File: backend/app/models/item.rb
class Item < ApplicationRecord
  include PublicIdGenerator
  include ShortInformation
  include LongInformation

  has_one_attached :image

  has_many :details, class_name: "ItemDetail"
  has_many :availabilities, class_name: "ItemAvailability"

  has_many :options
  has_many :variants

  has_many :relations 
  has_many :categories, through: :relations

  belongs_to :brand

  validates :public_id, presence: true, uniqueness: true
  validates :brand_id, presence: true
end

File: frontend/mobile/src/interfaces/Entities/Item.ts
export interface Item {
  public_id: string;
  name: string;
  brand: string;
  image: string;
  available: boolean;
  currency: string;
  options: {
    public_id: string;
    name: string;
    values: {
      value: string;
      unit: string | null;
    }[];
  }[];
  variants: {
    public_id: string;
    price: string;
    available: boolean;
    options: {
      public_id: string;
      name: string;
      value: string;
    }[];
  }[];
}
